mitodl/mitxpro
# Added top nav to static pages

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #104 

#### What's this PR do?
Adds a top nav bar to our static pages (course catalog, etc.)

#### How should this be manually tested?
Use the login/register/logout buttons. When you're logged in or logged out, click the logo in the top nav to navigate to the course catalog. The top nav should still be in the right state, and the buttons should behave as expected

#### Any background context you want to provide?
- I used the React context API to indicate whether the app was being run as a SPA or React mixed with Django templates. The solutions I saw for testing those context values were almost all broken, or they involved libraries that we don't use (jest). I settled for a semi-hacky-but-functional approach.
- Right now, the course catalog page (root url) does not load until the REST API call is made to fetch user info. That's just for simplicity's sake. We could render the bundle at the bottom and delay the API call until the rest of the page is rendered, but then we'd probably want some placeholder for the top nav bar.

#### Screenshots (if appropriate)
![ss 2019-04-10 at 15 25 35 ](https://user-images.githubusercontent.com/14932219/55907538-0ef24280-5ba5-11e9-8ebb-ee6eb1cf2098.png)
![ss 2019-04-10 at 15 25 42 ](https://user-images.githubusercontent.com/14932219/55907539-0ef24280-5ba5-11e9-80c3-37595e459f35.png)
![ss 2019-04-10 at 15 25 54 ](https://user-images.githubusercontent.com/14932219/55907540-0ef24280-5ba5-11e9-9310-8a6f5c7103b3.png)
![ss 2019-04-10 at 15 26 13 ](https://user-images.githubusercontent.com/14932219/55907541-0ef24280-5ba5-11e9-8af7-ba3e21f029ae.png)
